Sources/MlemMiddleware/API Client/ApiClient+Instance.swift

```diff
@@ -53,3 +53,29 @@ public extension ApiClient {
         }
     }
 }
+
+extension ApiClient {
+    /// Adds or removes an admin from this API's instance
+    func addAdmin(to instance_: any Instance3Providing, person: any Person, added: Bool) async throws {
+        let instance = instance_.instance3
+        
+        guard instance.api == self else {
+            assertionFailure("Instance API does not match self")
+            throw ApiClientError.invalidInput
+        }
+        
+        // the `to: instance` is a bit clunky but needed to set administrators without re-fetching the whole instance or
+        // asserting the presence of `myInstance`
+        let request = AddAdminRequest(personId: person.id, added: added)
+        let response = try await perform(request)
+        instance.administrators = await caches.person2.getModels(
+            api: self,
+            from: response.admins
+        )
```

Comment:
I don't think we need to require `instance` as an argument here. Would something like this work?

```suggestion
        let request = AddAdminRequest(personId: person.id, added: added)
        let response = try await perform(request)
        if let myInstance {
            myInstance.administrators = await caches.person2.getModels(
                api: self,
                from: response.admins
            )
        }
```

Reply:
I've implemented something a little different--this function now returns the `[Person2]` of the new admins, and it's up to the instance that calls it to update its own internal admin list.

Reply:
> this function is only called from an Instance3Providing

Currently, every other model-updating method in `ApiClient` (`voteOnPost`, `savePost` etc) is allowed to be called from outside of a middleware model and will update any existing middleware models itself. IMO we should still update the model here for consistency. Something like:

```swift
let administrators = caches.person2.getModels(
    api: self,
    from: response.admins
)
myInstance?.administrators = administrators
return administrators
```

Returning `[Person2]` is a good idea incase we need it, though it should be made `@discardableResult` if the method is changed to update the instance model.

Reply:
Also, it would be better for the function to accept a `personId: Int` rather than `person: any Person` for consistency with other `ApiClient` methods